lib/gui/etcher.ts

```diff
@@ -54,6 +54,27 @@ async function checkForUpdates(interval: number) {
 	}
 }
 
+const customProtocol = 'etcher';
+let openImageURL: string = process.argv.slice(1)[0] || '';
+
+// Take into account electron exec path argument for dev. env.
+if (!electron.app.isPackaged) {
+	openImageURL = process.argv.slice(2)[0] || '';
+}
+
+// This will catch clicks on links such as <a href="etcher://...">Open in Etcher</a>
+// We need to listen to the event before everything else otherwise the event won't be fired
+electron.app.on('open-url', async (event, data) => {
+	event.preventDefault();
+	openImageURL = data;
```

Comment:
There is no need to set a global.

Reply:
We do, otherwise OSX fires the event before the window is ready, this way we retain the value and emit the event to the renderer afterwards

Reply:
@thundron is it also passed in process.argv on macOS ?

Reply:
@zvin only when starting through command line, otherwise no, opening a URL from a browser just triggers the `open-url` event